Model/PaymentIntents.php

```diff
@@ -169,4 +166,27 @@ private function saveQuote(Quote $quote): void
             $this->quoteRepository->save($quote);
         }
     }
+
+    /**
+     * @param string $intentId
+     * @return mixed
+     * @throws GuzzleException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @throws \JsonException
+     */
+    public function cancelIntent(string $intentId)
+    {
+        try {
+            $response = $this->paymentIntentsCancel
+                ->setPaymentIntentId($intentId)
+                ->send();
+        } catch (GuzzleException $exception) {
+            $quote = $this->checkoutSession->getQuote();
+            $this->logger->quoteError($quote, 'intents', $exception->getMessage());
+            throw $exception;
+        }
+        $this->removeIntents();
```

Comment:
Seems like a code smell, whats the side-effect of $this->removeIntents?

Reply:
Intents are cached in magento internal cache. If we reload checkout the same wechat intent will be used. On method after intent cancelation we need to clear this cache otherwise the canceled intent will be passed in checkout. Let me know what should be changed here.